modules/tensor_mechanics/src/kernels/CrackPropagationHeatEnergy.C

```diff
@@ -0,0 +1,77 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "CrackPropagationHeatEnergy.h"
+#include "MooseMesh.h"
+#include "MooseVariable.h"
+
+template <>
+InputParameters
+validParams<CrackPropagationHeatEnergy>()
+{
+  InputParameters params = validParams<Kernel>();
+  params.addClassDescription("Crack propagation heat energy density = - (dPsi/dc) * (dc/dt) "
+                             "is the energy dissipated due to damage increase "
+                             "Psi is the free energy of the phase-field fracture model "
+                             "defined as Psi = (1 - c)^2 * G0_pos + G0_neg "
+                             "c is the order parameter for damage, continuous between 0 and 1 "
+                             "0 represents no damage, 1 represents fully cracked "
+                             "G0_pos and G0_neg are the positive and negative components "
+                             "of the specific strain energies");
+  params.addRequiredCoupledVar("c", "Phase field damage variable");
+  params.addRequiredCoupledVar("displacements",
+                               "The string of displacements suitable for the problem statement");
+  params.addParam<std::string>("base_name", "Material property base name");
+  return params;
+}
+
+CrackPropagationHeatEnergy::CrackPropagationHeatEnergy(const InputParameters & parameters)
+  : Kernel(parameters),
+    _base_name(isParamValid("base_name") ? getParam<std::string>("base_name") + "_" : ""),
+    _crack_propagation_heat(getMaterialProperty<Real>(_base_name + "crack_propagation_heat")),
+    _dcrack_propagation_heat_dstrain(
+        getMaterialProperty<RankTwoTensor>(_base_name + "dcrack_propagation_heat_dstrain")),
```

Comment:
Use `getMaterialPropertyDerivative()` here for these properties that contain derivatives. You'll need this class to derive from `DerivativeMaterialInterface<Kernel>` to do that.